### ISSUE TITLE
No error reported in the editor

### ISSUE THREAD

**Issue by wenfangdu** [microsoft/vscode-eslint#1251]
Error is reported in the terminal but is not reported in the editor:

![image](https://user-images.githubusercontent.com/28700378/117624109-23033680-b1a7-11eb-873a-72fec8056dce.png)

.eslintrc.js

```js
module.exports = {
  env: {
    node: true,
  },
  extends: [
    'plugin:vue/vue3-essential',
    'eslint:recommended',
    '@vue/typescript/recommended',
    '@vue/prettier',
    '@vue/prettier/@typescript-eslint',
  ],
  parserOptions: {
    ecmaVersion: 2020,
  },
  root: true,
  rules: {
    'object-shorthand': 'error',
  },
}
```

**Comment by dbaeumer** [comment]
@wenfangdu when validating in the terminal which working directory are you using. If it is different than VS Code's workspace folder you opened then very likely you need to configure `eslint.workingDirectories`. If it is the same then please provide me with a GitHub repository I can clone that demos that.

**Comment by wenfangdu** [comment]
@dbaeumer The working directories are the same, I've added repro [here](https://github.com/wenfangdu/vscode-eslint-repro), please have a look!

**Comment by dbaeumer** [comment]
I am a little puzzled with the example. What do you expect in main.ts to happen. If I change the code to 

```
import { createApp } from "vue";
import App from "./App.vue";

createApp(App).mount("#app");

const a = "a";
// 👇 ts is not
console.log({ a: a });
```

to mirror what is in test.js I get the correct error

<img width="515" alt="capture" src="https://user-images.githubusercontent.com/1931590/117672716-60b58e80-b1aa-11eb-900a-f9f0ed3ddba8.png">

**Comment by wenfangdu** [comment]
@dbaeumer Oops, sorry, seems `vue-cli-service lint` fixed `main.ts` issue when committing, I've recovered the code in `main.ts` to the following ([repro](https://github.com/wenfangdu/vscode-eslint-repro) is updated), mine is not throwing 🤔:

![image](https://user-images.githubusercontent.com/28700378/117735021-4077e500-b227-11eb-9b5e-70416171d14e.png)

**Comment by wenfangdu** [comment]
@dbaeumer I've fixed this issue by adding `parser: "@typescript-eslint/parser"` into `.eslintrc.js`, the codebase was generated by `vue-cli`, it didn't generate this line. Have you added `parser` before trying, if you didn't, then this issue is weirder 🤔.

**Comment by wenfangdu** [comment]
@dbaeumer  I seem to have found the cause and open a new issue https://github.com/microsoft/vscode-eslint/issues/1266, when using `vue-eslint-parser`, and with `"eslint.validate": []`, `.ts` and `.tsx` files aren't auto-detected, unless I explicitly specify:

```json
{
  "eslint.validate": [
    "javascript",
    "javascriptreact",
    "typescript",
    "typescriptreact"
  ]
}
```
